lib/backup/Backup.ts

```diff
@@ -0,0 +1,177 @@
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import { createHash } from 'crypto';
+import Config from '../Config';
+import Logger, { Context } from '../Logger';
+import LndClient from '../lndclient/LndClient';
+
+class Backup {
+  private logger!: Logger;
+  private config = new Config();
+
+  private backupDir!: string;
+
+  private fileWatchers: fs.FSWatcher[] = [];
+  private lndClients: LndClient[] = [];
+
+  public start = async (args: { [argName: string]: any }) => {
+    await this.config.load(args);
+
+    this.backupDir = args.backupdir || this.getDefaultBackupDir();
+
+    this.logger = new Logger({
+      context: Context.Backup,
+      level: this.config.loglevel,
+      filename: this.config.logpath,
+      instanceId: this.config.instanceid,
+      dateFormat: this.config.logdateformat,
+    });
+
+    if (!fs.existsSync(this.backupDir)) {
+      try {
+        fs.mkdirSync(this.backupDir);
+      } catch (error) {
+        this.logger.error(`Could not create backup directory: ${error}`);
+        return;
+      }
+    }
+
+    try {
+      await this.startLndSubscriptions();
+    } catch (error) {
+      this.logger.error(`Could not connect to LNDs: ${error}`);
+    }
+
+    // Start the Raiden database filewatcher
+    if (args.raiden) {
+      this.startFilewatcher('raiden', args.raiden.database);
+    } else {
+      this.logFileNotSpecified('raiden');
+    }
+
+    // Start the XUD database filewatcher
+    if (args.dbpath !== undefined && args.dbpath !== '') {
```

Comment:
@erkarl and I are thinking it would make sense not to reference `args` here (and above for raiden) but rather `this.config`. That way we can use values from the config file or default values and not have to pass in all of these values via command line argument. That might make the `logFileNotSpecified` function obsolete since there will always be a default value.